src/main/flight/pid.c

```diff
@@ -545,8 +532,8 @@ FAST_CODE void pidController(const pidProfile_t *pidProfile, const rollAndPitchT
             axisPID_D[axis] = Kd[axis] * delta * tpaFactor;
         }
 
-        // Disable PID control at zero throttle
-        if (!pidStabilisationEnabled) {
+        // Disable PID control if at zero throttle or if gyro overflow detected
+        if (!pidStabilisationEnabled || gyroOverflowDetected()) {
```

Comment:
I-terms are still accumulating (possibly using inverted values)

Reply:
@ledvinap , I don't think the I-terms are actually accumulating. What I think is happening is that because the "drill of doom" does not respond to the PID output, the I-term is just doing it's job - accumulating the uncorrected error.

Reply:
What is gyro is inverted for long enough ? It will accumulate in opposite direction, speeding the spin up when yaw rate gets under threshold ... (But gyro reading will be correct from 4kdeg/s, so I-term should have enough time to settle).

Yaw will be stopped with maximum I-action, which may cause problems(bigger nonlinearity). It will probably cause overshoot (unless D term slows it down).

roll and pitch I-term accumulate too. With quad spinning fast, it may even lead to PID coupling and another flyaway ..

And in general, it's better to avoid hard nonlinearities where possible - it may be even good idea so ramp control signals when spin is slowing ...